fcrepo-persistence-ocfl/src/main/java/org/fcrepo/persistence/ocfl/impl/IndexBuilderImpl.java

```diff
@@ -77,10 +95,28 @@ public void rebuild() {
                             //we're only interested in sidecar subpaths
                             try {
                                 final var headers = deserializeHeaders(objSession.read(subpath));
-                                fedoraIds.add(headers.getId());
+                                final var fedoraId = headers.getId();
+                                fedoraIds.add(fedoraId);
                                 if (headers.isArchivalGroup()) {
                                     rootId.set(headers.getId());
                                 }
+
+                                if (!fedoraId.equals(rootFedoraId)) {
```

Comment:
If you create `fedoraId` as a `FedoraId` then you could use `fedoraId.isRepositoryRoot()` here.

Reply:
I'll definitely do that.